src/frontend/src/planner/relation.rs

```diff
@@ -57,13 +57,10 @@ impl Planner {
     }
 
     pub(crate) fn plan_sys_table(&mut self, sys_table: BoundSystemTable) -> Result<PlanRef> {
-        Ok(LogicalScan::create(
+        Ok(LogicalSysScan::create(
             sys_table.sys_table_catalog.name().to_string(),
-            ScanTableType::SysTable,
             Rc::new(sys_table.sys_table_catalog.table_desc()),
```

Comment:
`SystemTableCatalog::table_desc` generates a `TableDesc` with most fields being dummy. I guess we should take a more native structure here and it's decoupled with normal table scan.

Reply:
Add it later, requires further refactoring of other common interfaces, e.g. `split_to_scan_ranges`.